client/src/components/MyProfile.js

```diff
@@ -1,9 +1,272 @@
+    setEditedPhone(PhoneHistory);
+    setEditedAddress(AddressHistory);
+    setEditedPostalcode(PostalcodeHistory);
+    setEditedCity(CityHistory);
+    setEditedCountry(CountryHistory);
+    setEditedEmail(EmailHistory);
+
+    // Close the edit
+    setEdit(false);
+  }
+
+  const handleFirstnameChange = (event) =>
+  {
+    setEditedFirstname(event.target.value);
+  }
+
+  const handleLastnameChange = (event) =>
+  {
+    setEditedLastname(event.target.value);
+  }
+
+  const handlePhoneChange = (event) =>
+  {
+    setEditedPhone(event.target.value);
+  }
+
+  const handleAddressChange = (event) =>
+  {
+    setEditedAddress(event.target.value);
+  }
+
+  const handlePostalcodeChange = (event) =>
+  {
+    setEditedPostalcode(event.target.value);
+  }
+
+  const handleCityChange = (event) =>
+  {
+    setEditedCity(event.target.value);
+  }
+
+  const handleCountryChange = (event) =>
+  {
+    setEditedCountry(event.target.value);
+  }
+
+  const handleEmailChange = (event) =>
+  {
+    setEditedEmail(event.target.value);
+  }
 
-function MyProfile() {
   return (
-    <div>MyProfile</div>
+    <div className='MyInfo'>
+      <img src='https://blogs.lut.fi/newcomers/wp-content/uploads/sites/15/2020/02/talvi-ilma-1-1.jpg' />
+      <div>
+        <p>Firstname: {Firstname}</p>
+        <p>Lastname: {Lastname}</p>
+        <p>Phone: {Phone}</p>
+        <p>Address: {Address}</p>
+        <p>Postalcode: {Postalcode}</p>
+        <p>City: {City}</p>
+        <p>Country: {Country}</p>
+      </div>
+      <Button variant='outlined' onClick={editOnClick} >Edit</Button>
+
+    <EditDetailsModal 
+      edit={edit} 
+      editedEmail={editedEmail} 
+      editedCountry={editedCountry}
+      editedCity={editedCity}
+      editedPhone={editedPhone}
+      editedAddress={editedAddress}
+      editedPostalcode={editedPostalcode}
+      handleFirstnameChange={handleFirstnameChange}
+      handleLastnameChange={handleLastnameChange}
+      handlePhoneChange={handlePhoneChange}
+      handleAddressChange={handleAddressChange}
+      handlePostalcodeChange={handlePostalcodeChange}
+      handleCityChange={handleCityChange}
+      handleCountryChange={handleCountryChange}
+      handleEmailChange={handleEmailChange}
+      cancelEditOnClick={cancelEditOnClick}
+      saveEditOnClick={saveEditOnClick}
+    />
+    </div>
+  )
+}
+
+
+
+function Home() {
+  const [user, setUser ] = useState(null);
+  const userId = sessionStorage.getItem('id');
+
+  const fetchUserId = async () => {
+    try {
+      const response = await fetch('http://localhost:4000/api/getID');
+      if (!response.ok) {
+        throw new Error('Failed to fetch user data');
+      }
+      sessionStorage.setItem("id",await response.json());
+    } catch (error) {
+      console.error('Error fetching user data:', error);
+    }
+  };
+
+  const fetchUserData = async () => {
+    try {
+      const response = await fetch(`http://localhost:4000/users/getData/${userId}`);
```

Comment:
Url

Reply:
I don't know why, but without the localhost the fetch doesnt work, none of them.

Reply:
Have you set the environment variables in client/.env ? It should contain
`REACT_APP_PROXY_TARGET='http://localhost:4000'

😄

Reply:
Has this been fixed?

Reply:
I think so